types/types.go

```diff
@@ -115,7 +116,7 @@ type ImageSource interface {
 	// out of a manifest list.
 	GetTargetManifest(digest string) ([]byte, string, error)
 	// GetBlob returns a stream for the specified blob, and the blob’s size (or -1 if unknown).
-	GetBlob(digest string) (io.ReadCloser, int64, error)
+	GetBlob(BlobInfo) (io.ReadCloser, int64, error)
```

Comment:
(AFAICT we don’t need it, so this would be just a matter of documenting `The Digest field is guaranteed to be provided; Size may be -1` just as we do for `LayerInfos` where this information comes from.)